tests/unittest_scoped_nodes.py

```diff
@@ -466,6 +466,77 @@ def func():
         self.assertIsInstance(func_vals[0], nodes.Const)
         self.assertIsNone(func_vals[0].value)
 
+    def test_no_returns_is_implicitly_none(self):
+        code = """
+            def f():
+                pass
+            value = f()
+            value
+        """
+        node = builder.extract_node(code)
+        inferred = next(node.infer())
+        assert isinstance(inferred, nodes.Const)
+        assert inferred.value is None
+
+    def test_only_raises_is_not_implicitly_none(self):
+        code = """
+            def f():
+                raise SystemExit()
+            f()
+        """
+        node = builder.extract_node(code)  # type: nodes.Call
+        inferred = next(node.infer())
+        assert inferred is util.Uninferable
+
+    def test_abstract_methods_are_not_implicitly_none(self):
+        code = """
+            from abc import ABCMeta, abstractmethod
+
+            class Abstract(metaclass=ABCMeta):
+                @abstractmethod
+                def foo(self):
+                    pass
+                def bar(self):
+                    pass
+            Abstract().foo()  #@
+            Abstract().bar()  #@
+
+            class Concrete(Abstract):
+                def foo(self):
+                    return 123
+            Concrete().foo()  #@
+            Concrete().bar()  #@
+        """
+        afoo, abar, cfoo, cbar = builder.extract_node(code)
+
+        assert next(afoo.infer()) is util.Uninferable
+        for node, value in [(abar, None), (cfoo, 123), (cbar, None)]:
+            inferred = next(node.infer())
+            assert isinstance(inferred, nodes.Const)
+            assert inferred.value == value
+
+    @pytest.mark.xfail(reason="unimplemented")
```

Comment:
Sure, I'll add a comment explaining this in the code.

Reply:
I've forgotten some of the reasoning behind this, but I'm fairly sure it was about ignoring "abstract" functions, since the change around implicit return `None` caused `pylint` to fail on the `astroid` source code for basically this reason: `NodeNG` has a stub `_infer` method that everything else overloads. I can't remember what this test was *supposed* to look like though, and it passes now so there's no real point to it until I can remember what it meant to look like (like multiple inferred return values for each subclass etc.) so I'll remove it.